### ISSUE TITLE
Getting FreeHeap Memory

### ISSUE THREAD

**Issue by tqrahman** [Call-for-Code/ClusterDuck-Protocol#196]
To get heap memory I added this to the Duck.h file.
`int getFreeHeapMemory() { return duckesp::freeHeapMemory(); };`
and in the application layer I add 
`int heap = duck.getFreeHeapMemory();`
For some reason I do not get the right value. The heap memory increases every time runSensor() is called by the INTERVAL_MS i set . In the example I set INTERVAL_MS to 1000, so the heap memory increases by 1000. If i set it to 5000, the heap memory increases by 5000. 

<img width="260" alt="Screen Shot 2021-05-11 at 12 57 43 PM" src="https://user-images.githubusercontent.com/17665316/117877398-5b634d00-b259-11eb-853e-41feccf6e072.png">

**Comment by tqrahman** [comment]
when I get the heap memory from ESP, then it works fine.
`int heap = ESP.getFreeHeap();`

**Comment by amirna2** [comment]
Have you tried the one provided in CDP MemoryFree.h? It works for both ARM and ESP32 chips.

**Comment by tqrahman** [comment]
How can I implement MemoryFree.h correctly? 

In the Custom MamaDuck example, I included: 
`#include <MemoryFree.h>`
and added freeMemory() function in the runSensor function: 
`String message = String("Counter:") + String(counter) + String(" Heap: ") + String(freeMemory());`

And I only get 1's as output.

<img width="621" alt="Screen Shot 2021-06-23 at 5 16 21 AM" src="https://user-images.githubusercontent.com/17665316/123071351-9624dd00-d3e2-11eb-877c-aad6a8f550fb.png">

**Comment by amirna2** [comment]
What board is your MamaDuck?
I tried on my Heltec ESP32 WiFi v2 PapaDuck and it seems to be working:

` Serial.println( "[PAPA] Free Memory: " + String(freeMemory()) );`

```
21:21:43.604 -> [PAPA] Publish ok
21:21:43.644 -> [PAPA] Free Memory: 189024
```

**Comment by tqrahman** [comment]
Yeah I am using the CustomMama example with board Heltec ESP32 WiFi v2. Now I added an extra line:
```
int heap = freeMemory();
String message = String("Counter:") + String(counter) + String(" Heap:") + heap;
```
Now getting: 
`[MAMA] sensor data: Counter:1 Heap:491388928`
which seems too high.

**Comment by nfeuer** [comment]
@tqrahman can you try this again and confirm if this is still an issue? @faradaym did some work on the free heap stuff #238

**Comment by tqrahman** [comment]
this issue is resolved with the [PR#323](https://github.com/Call-for-Code/ClusterDuck-Protocol/pull/323)
